diff --git a/dlib/error.h b/dlib/error.h
index ce9b95b1..f2203cc9 100644
--- a/dlib/error.h
+++ b/dlib/error.h
@@ -71,7 +71,7 @@ namespace dlib
         error(
             error_type t,
             const std::string& a
-        ): info(a), type(t) {}
+        ): info(a), type(t) {dump_error();}
         /*!
             ensures
                 - #type == t
@@ -80,7 +80,7 @@ namespace dlib
 
         error(
             error_type t
-        ): type(t) {}
+        ): type(t) {dump_error();}
         /*!
             ensures
                 - #type == t
@@ -89,7 +89,7 @@ namespace dlib
 
         error(
             const std::string& a
-        ): info(a), type(EUNSPECIFIED) {}
+        ): info(a), type(EUNSPECIFIED) {dump_error();}
         /*!
             ensures
                 - #type == EUNSPECIFIED
@@ -97,7 +97,7 @@ namespace dlib
         !*/
 
         error(
-        ): type(EUNSPECIFIED) {}
+        ): type(EUNSPECIFIED) {dump_error();}
         /*!
             ensures
                 - #type == EUNSPECIFIED
@@ -175,6 +175,15 @@ namespace dlib
 
     private:
         const error& operator=(const error&);
+
+        void dump_error()
+        {
+            std::cerr << "\n**************************** " << type_to_string() << " ERROR DETECTED ****************************\n";
+            for (unsigned int i = 0; i < this->info.size(); ++i) {
+               std::cerr << info[i];
+            }
+            std::cerr << "\n******************************************************************************\n\n";
+        }
     };
 
 // ----------------------------------------------------------------------------------------
